Source/WebCore/animation/AnimationTimelinesController.cpp

```diff
@@ -314,13 +316,112 @@ void AnimationTimelinesController::maybeClearCachedCurrentTime()
         m_cachedCurrentTime = std::nullopt;
 }
 
+static Element* originatingElement(const Ref<ScrollTimeline>& timeline)
+{
+    if (RefPtr viewTimeline = dynamicDowncast<ViewTimeline>(timeline))
+        return viewTimeline->subject();
+    return timeline->source();
+}
+
+static Element* originatingElementIncludingTimelineScope(const Ref<ScrollTimeline>& timeline)
+{
+    if (WeakPtr element = timeline->timelineScopeDeclaredElement())
+        return element.get();
+    return originatingElement(timeline);
+}
+
+static Element* originatingElementExcludingTimelineScope(const Ref<ScrollTimeline>& timeline)
+{
+    return timeline->timelineScopeDeclaredElement() ? nullptr : originatingElement(timeline);
+}
+
+Vector<WeakPtr<Element, WeakPtrImplWithEventTargetData>> AnimationTimelinesController::relatedTimelineScopeElements(const AtomString& name)
+{
+    Vector<WeakPtr<Element, WeakPtrImplWithEventTargetData>> timelineScopeElements;
+    for (auto& scope : m_timelineScopeEntries) {
+        if (scope.second && (scope.first.type == TimelineScope::Type::All || (scope.first.type == TimelineScope::Type::Ident && scope.first.scopeNames.contains(name))))
+            timelineScopeElements.append(scope.second);
+    }
+    return timelineScopeElements;
+}
+
+static ScrollTimeline* determineTreeOrder(Vector<Ref<ScrollTimeline>> ancestorTimelines, const Element& matchElement, Vector<WeakPtr<Element, WeakPtrImplWithEventTargetData>> timelineScopeElements)
+{
+    RefPtr element = &matchElement;
+    while (element) {
+        Vector<Ref<ScrollTimeline>> matchedTimelines;
+        for (auto& timeline : ancestorTimelines) {
+            if (element == originatingElementIncludingTimelineScope(timeline))
+                matchedTimelines.append(timeline);
+        }
+        if (!matchedTimelines.isEmpty()) {
+            if (timelineScopeElements.contains(element.get())) {
+                if (matchedTimelines.size() == 1)
+                    return matchedTimelines.first().ptr();
+                // Conflict due to timeline-scope
```

Comment:
Still not sure what this comment means.

Reply:
This means that multiple elements could be associated due to timeline-scope, which results in an inactive timeline.

Reply:
This would have been a better comment to put in the commit.